torch/fx/_pytree.py

```diff
@@ -0,0 +1,37 @@
+from typing import Callable, Any, Tuple, List, Dict, Type
+from torch.utils._pytree import PyTree, TreeSpec, LeafSpec
+
+FlattenFuncSpec = Callable[[PyTree, TreeSpec], List]
+
+SUPPORTED_NODES: Dict[Type[Any], Any] = {}
+def _register_pytree_flatten_spec(typ: Any, flatten_fn_spec: FlattenFuncSpec) -> None:
+    SUPPORTED_NODES[typ] = flatten_fn_spec
+
+def tree_flatten_spec(pytree: PyTree, spec: TreeSpec) -> List[Any]:
+    if isinstance(spec, LeafSpec):
+        return [pytree]
+    if spec.type not in SUPPORTED_NODES:
+        raise RuntimeError(
+            f"{type(pytree)} does not have a flatten_fn_spec associated with it. Please register one with"
+            "torch.fx._pytree._register_pytree_flatten_spec.  If you have serialized your model, make"
```

Comment:
Seems weird to be pointing the user to call internal APIs. If the pytree module is private because it's not ready, I would at least expect the APIs inside of it (e.g. `_register_pytree_flatten_spec`) to be public in anticipation of making the module public